addons/etcd/scripts/switchover-with-candidate.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
+
+set -ex
+leaderEndpoint=${KB_LEADER_POD_FQDN}.${KB_NAMESPACE}.svc${CLUSTER_DOMAIN}:2379
+candidateEndpoint=${KB_SWITCHOVER_CANDIDATE_FQDN}.${KB_NAMESPACE}.svc${CLUSTER_DOMAIN}:2379
+
+# see common.sh, this function may change leaderEndpoint
+updateLeaderIfNeeded 3
+
+if [ "$leaderEndpoint" = "$candidateEndpoint" ]; then
+  echo "leader is the same as candidate, no need to switch"
+  exit 0
+fi
+
+candidateID=$(execEtcdctlNoCheckTLS ${candidateEndpoint} endpoint status | awk -F', ' '{print $2}')
+execEtcdctlNoCheckTLS ${leaderEndpoint} move-leader $candidateID
```

Comment:
It is recommended to specify all node addresses in the etcd cluster using the --endpoints option when using the etcdctl command. This improves command reliability.

Reply:
https://github.com/apecloud/kubeblocks/issues/7950
TODO in the future